client/extensions/woocommerce/app/settings/payments/stripe/payment-method-stripe-connect-account.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Image from 'components/image';
+import resizeImageUrl from 'lib/resize-image-url';
+
+class StripeConnectAccount extends Component {
+	static propTypes = {
+		stripeConnectAccount: PropTypes.shape( {
+			displayName: PropTypes.string,
+			email: PropTypes.string,
+			firstName: PropTypes.string,
+			isActivated: PropTypes.bool,
+			lastName: PropTypes.string,
+			logo: PropTypes.string,
+		} ),
+		onDisconnect: PropTypes.func, // TODO - require most of these props in subsequent PR
+	}
+
+	renderLogo = () => {
+		const { stripeConnectAccount } = this.props;
+		const { logo } = stripeConnectAccount;
+
+		let image = null;
+
+		if ( ! isEmpty( logo ) ) {
+			image = <Image src={ resizeImageUrl( logo, { w: 40, h: 40 } ) } className="stripe__connect-account-logo" />;
+		}
+
+		return (
+			<div className="stripe__connect-account-logo-container">
+				{ image }
+			</div>
+		);
+	}
+
+	renderNameAndEmail = () => {
+		const { stripeConnectAccount } = this.props;
+		const { displayName, email, firstName, lastName } = stripeConnectAccount;
+		const name = ! isEmpty( displayName ) ? displayName : `${ firstName } ${ lastName }`;
+
+		return (
+			<div className="stripe__connect-account-details">
+				<span className="stripe__connect-account-name">
+					<span className="stripe__connect-account-name">{ name }</span>
+					<span className="stripe__connect-account-email">{ email }</span>
+				</span>
+			</div>
+		);
+	}
+
+	// TODO - when we are ready to connect this for-reals, this layer may not be needed
+	onDisconnect = ( event ) => {
+		event.preventDefault();
+		if ( this.props.onDisconnect ) {
+			this.props.onDisconnect();
+		}
+	}
+
+	renderStatus = () => {
+		const { stripeConnectAccount, translate } = this.props;
+		const { isActivated } = stripeConnectAccount;
+
+		let status = null;
+
+		if ( isActivated ) {
+			status = <span className="stripe__connect-account-status account-activated">
```

Comment:
Seems like this if/else block could be simplified a bit... maybe the css class for `stripe__connect-account-status` could default to have a green "active" text, and perhaps `account-not-activated` class name could be `is-not-activated` to trigger the red text, then we could have something like the following on line 86

```
<span className={ classNames( 'stripe__connect-account-status', { 'is-not-activated': isActivated } ) }>
    { isActivated ?
        { translate( 'Activated' ) } :
        { translate( 'Check email to activate account' ) }
    }
</span>
```

Reply:
I can see both sides of this. On one hand, @timmyc, your version is more terse and compact, but on the other hand, it's a bit more difficult to visually parse.